cmake/compilerFlags.cmake

```diff
@@ -1,7 +1,7 @@
 # These flags applies to exiv2lib, the applications, and to the xmp code
 include(CheckCXXCompilerFlag)
 
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 14)

```

Comment:
Very much in favor of bumping this up, but any chance we can make the jump all the way to 17?  

I'd be quite interested in the deduction guides for `std::array` which would enable us to make a lot of the makernote code quite a bit more concise, readable, and safer :blush:

Reply:
I'm OK with changing this.  Please don't go beyond 2014 because that's the highest version that 0.27-maintenance can support.  If we jump to 2020, we'll leave the 2017 people without a build.

My C++ skills are frozen in time around 2010.  I don't know anything much about 11, 14, or 17.  I read an interview with Bjarne Stroudstroup and he was saying that C++20 is the language he hoped to achieve from the very beginning.  Maybe it's worth thinking that after `0.27-maintenance` is finished and history, 'main' might be the elderly branch and a 'modern' uses C++20 aggressively.

Reply:
I'm not sure if I understand the point about `0.27-maintenance`, that branch is fixed on `c++98` right?  
So,  how does the version we use on the `main` branch interact with the maintenance branch?

Reply:
@hassec.  Apologies.  I am confusing you.  I'm thinking we currently have two major branches. The legacy 98/0.27 and 'main'.  I hope`0.27-maintenance` will soon be totally closed.  And I promise to retire with that branch.

I'm OK with 'main' requiring C++17.

Maybe C++20 is a new horizon which deserves a 'modern' branch.

Reply:
For me it would be also fine to upgrade here to C++17. As we discusses few years ago, the main limitation fact for me is the default C++ compiler available in each major Linux distribution (at least the Linux distributions we are testing on CI).

I will investigate the upgrade to C++17 in a separate PR. In this one, I have already other things to investigate :smiling_face_with_tear:

Reply:
I tried to do the upgrade to C++17 here, but it is not as simple as just changing the CMake code. We need to make some fixes in the codebase too. So I will try that in a different branch.